### PR TITLE
RPC: unify return type for ExecuteTransaction

### LINKED ISSUE #3854: RPC: unify the return type for `executeTransaction`
Right now we are returning [different TransactionResponse](https://github.com/MystenLabs/sui/blob/836d76c289c798187a596b024a335bcc86adafdf/crates/sui-core/src/gateway_state.rs#L1224) variant for each transaction kind. This actually makes things harder to parse on the client side, which has to [do a lot of extra work](https://github.com/MystenLabs/sui/blob/main/sdk/typescript/experimental/src/rpc.ts#L172) to parse transaction response. We should make the API always return `TransactionResponse::EffectResponse`. 

The `sui client` CLI can do some extra work of transforming the requests if it wants to show a more succinct message.

### PR DESCRIPTION
**Pull request by 666lcz**
The key change of this PR is in crates/sui-json-rpc-types/src/lib.rs

## Before

`execute_transaction` returns `TransactionResponse`, the schema of which varies between different transaction types

```
#[derive(Serialize, Deserialize, Debug, JsonSchema)]
pub enum TransactionResponse {
    EffectResponse(TransactionEffectsResponse),
    PublishResponse(PublishResponse),
    MergeCoinResponse(MergeCoinResponse),
    SplitCoinResponse(SplitCoinResponse),
}
```
## After

`execute_transaction` now returns `SuiTransactionEffectsResponse`, which will always have `certificate`, `effects`, `timestamp_ms` field. I added an optional `parsed_data` field for convenience. Clients can choose to ignore this field(i.e., it does not make client side parsing harder), but some clients(e.g., sui client cli) may find the parsed data helpful. For example, for Publish Transaction, the parsed_data field may contain a `package_id` field. Without this field, the client would have to make additional queries for each of the created objects to find out their types.

```
#[derive(Serialize, Deserialize, Debug, JsonSchema)]
pub struct SuiTransactionEffectsResponse {
    pub certificate: SuiCertifiedTransaction,
    pub effects: SuiTransactionEffects,
    pub timestamp_ms: Option<u64>,
    pub parsed_data: Option<SuiParsedTransactionResponse>,
}

#[derive(Serialize, Deserialize, Debug, JsonSchema, Clone)]
pub enum SuiParsedTransactionResponse {
    Publish(SuiParsedPublishResponse),
    MergeCoin(SuiParsedMergeCoinResponse),
    SplitCoin(SuiParsedSplitCoinResponse),
}
```

## Testing

In addition to the unit test, I also manually tested the `sui client` as the `Display` trait implementation is changed. Note that the parsed data such as `Split Coin Results` are still rendered.

```
➜ ./sui client split-coin --amounts 100 --coin-id 0x043e5935d0bd263b91de724d6f58a46b038a100b --gas-budget 1000
----- Certificate ----
Transaction Hash: re8jFpLqcQQAVkze3pE5j9JRRDA/mxgqOvufnpWvoG8=
Transaction Signature: AA==@AaW+Oy4xPVVChb4lFHxicsQe9gpWwP2Pgxn4w25z2UNAkujXjEAdHeDzGUqyQq8cPBY2dOWlJlATfQZJ5m21Cg==@UoCg/5a0f/fbC1If8AofBRAs4wSHpuzkpUbhCF7dn8I=
Signed Authorities Bitmap: RoaringBitmap<[0, 1, 2]>
Transaction Kind : Call
Package ID : 0x2
Module : coin
Function : split_vec
Arguments : ["0x43e5935d0bd263b91de724d6f58a46b038a100b", [100]]
Type Arguments : ["0x2::sui::SUI"]
----- Transaction Effects ----
Status : Success
Created Objects:
  - ID: 0xf3476a213c021166c19eab8444e43b025549724b , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
Mutated Objects:
  - ID: 0x043e5935d0bd263b91de724d6f58a46b038a100b , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
  - ID: 0x42aac345cc504060cab63c398bf2e4750e682b18 , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
----- Split Coin Results ----
Updated Coin : Coin { id: 0x043e5935d0bd263b91de724d6f58a46b038a100b, value: 99999900 }
New Coins : Coin { id: 0xf3476a213c021166c19eab8444e43b025549724b, value: 100 }
Updated Gas : Coin { id: 0x42aac345cc504060cab63c398bf2e4750e682b18, value: 99999473 }
```
```
➜ ./sui client merge-coin --primary-coin 0xaef3f3c497ccf44f58ae49e5f6d296650c909727 --coin-to-merge 0x97f88e91be1b5cd9ab4b2d9257af0867eae6d474 --gas-budget 1000
----- Certificate ----
Transaction Hash: 6nkRyCP1OViLnavKhRw4dvBagw9O5zr9CauIFatoPEU=
Transaction Signature: AA==@CdKTVbjNiZCaRzj/2gIZAI4xwBOG+AClMQzUbEn0wUP8UaBwWMzTCDqolRq8z+lGVaSykMQgoRJf/vV8LpyXAg==@UoCg/5a0f/fbC1If8AofBRAs4wSHpuzkpUbhCF7dn8I=
Signed Authorities Bitmap: RoaringBitmap<[0, 1, 2]>
Transaction Kind : Call
Package ID : 0x2
Module : coin
Function : join
Arguments : ["0xaef3f3c497ccf44f58ae49e5f6d296650c909727", "0x97f88e91be1b5cd9ab4b2d9257af0867eae6d474"]
Type Arguments : ["0x2::sui::SUI"]
----- Transaction Effects ----
Status : Success
Mutated Objects:
  - ID: 0x043e5935d0bd263b91de724d6f58a46b038a100b , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
  - ID: 0xaef3f3c497ccf44f58ae49e5f6d296650c909727 , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
Deleted Objects:
  - ID: 0x97f88e91be1b5cd9ab4b2d9257af0867eae6d474
----- Merge Coin Results ----
Updated Coin : Coin { id: 0xaef3f3c497ccf44f58ae49e5f6d296650c909727, value: 200000000 }
Updated Gas : Coin { id: 0x043e5935d0bd263b91de724d6f58a46b038a100b, value: 99999429 }
```

```
fastnft/target/debug on  chris-unify-transaction-response [$!] using ☁️  default/sui-devnet
➜ ./sui client publish --path ~/projects/toy-move-packages/fungible-tokens/ --gas-budget 30000
----- Certificate ----
Transaction Hash: 3RbcdT8t6YPTfKxXon5RQdmbDWQwjHILUx3LVUU8HGU=
Transaction Signature: AA==@TBwXTCVBacpnRgUsz19EoSbXk+HVcq0I+ZhrGDSLbIyyr/3cQBBZ19+I+OFxu/1v2k8dprU/gQnP6XUMOVhMDQ==@UoCg/5a0f/fbC1If8AofBRAs4wSHpuzkpUbhCF7dn8I=
Signed Authorities Bitmap: RoaringBitmap<[1, 2, 3]>
Transaction Kind : Publish
----- Transaction Effects ----
Status : Success
Created Objects:
  - ID: 0x816726a6501f709d202010a46bea2ccb284f3540 , Owner: Immutable
  - ID: 0xab0ece644e06f2b25a96821f77483d503c928812 , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
Mutated Objects:
  - ID: 0x043e5935d0bd263b91de724d6f58a46b038a100b , Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
----- Publish Results ----
The newly published package object ID: 0x816726a6501f709d202010a46bea2ccb284f3540

List of objects created by running module initializers:
----- Move Object (0xab0ece644e06f2b25a96821f77483d503c928812[1]) -----
Owner: Account Address ( 0x03530db0719259f8b5ff912d05e020b2ab40745e )
Version: 1
Storage Rebate: 16
Previous Transaction: 3RbcdT8t6YPTfKxXon5RQdmbDWQwjHILUx3LVUU8HGU=
----- Data -----
type: 0x2::coin::TreasuryCap<0x816726a6501f709d202010a46bea2ccb284f3540::usdc::USDC>
id: 0xab0ece644e06f2b25a96821f77483d503c928812
total_supply:
  type: 0x2::balance::Supply<0x816726a6501f709d202010a46bea2ccb284f3540::usdc::USDC>
  value: 0

Updated Gas : Coin { id: 0x043e5935d0bd263b91de724d6f58a46b038a100b, value: 99998848 }

```

## Checklist

- [x]  Update TS-SDK
- [x]  Update Explorer
- [x]  Update Wallet Browser Extension

#closes #3854

### CODE REVIEW

**Comment by damirka** [comment]
Dumb question - do we return `package` in transaction effects? That's a must.

**Review by damirka — APPROVED** [review]
LTGM but I'm not the best person to review Rust bits.

File: crates/sui-json-rpc-types/src/lib.rs [hunk 1]
```diff
@@ -303,55 +303,56 @@ pub enum MoveFunctionArgType {
 }
 
 #[derive(Serialize, Deserialize, Debug, JsonSchema)]
-pub struct TransactionEffectsResponse {
+pub struct SuiTransactionEffectsResponse {
```
Comment (patrickkuo) [inline crates/sui-json-rpc-types/src/lib.rs:306]: can we rename this to TransactionResponse/SuiTransactionResponse? Now effects is the only response type, we don't really need it in the name.

File: crates/sui-json-rpc-types/src/lib.rs [hunk 2]
```diff
@@ -303,55 +303,56 @@ pub enum MoveFunctionArgType {
...
     pub certificate: SuiCertifiedTransaction,
     pub effects: SuiTransactionEffects,
     pub timestamp_ms: Option<u64>,
+    pub parsed_data: Option<SuiParsedTransactionResponse>,
```
Comment (patrickkuo) [inline crates/sui-json-rpc-types/src/lib.rs:310]: If this is only useful to the Sui CLI, I think we can remove this in future PR after my client side object state PR is done. Converting effect to specialised response type on client side should be cheap after that.

**Review by patrickkuo — APPROVED** [review]
Nice one! just a minor comment on naming.

The changes in json sample files are really quite annoying and makes the PR look huge lol, do we still need them for TS SDK testing? If we still need them, I am thinking maybe we can make the test network generate deterministic values by using seeded Rng.

**Comment by 666lcz** [comment]
> The changes in json sample files are really quite annoying and makes the PR look huge lol, do we still need them for TS SDK testing? 

We should be able to get rid of those pretty soon as @Jordan-Mysten is moving the test data generation to the client side.

File: crates/sui-json-rpc-types/src/lib.rs (hunk 1 above)
Comment (666lcz) [reply crates/sui-json-rpc-types/src/lib.rs]: Done!

**Pull request merged**